src/components/viewerJs/viewerJsNativeOverrides.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * PDF viewer aware native overrides.
+ * This file contains a set of overrides, customised to work with a viewer.js based PDF viewer component, that will 
+ * convert HTML5 window actions to corresponding Finsemble actions.
+ * Overrides must be specified for each component via a "preload" script. You do this by adding the preload to the
+ * component config like so:
+ * 
+ * 	"Your Component": {
+  			...
+			"component": {
+				...
+				"inject": false,
+				"preload": "$applicationRoot/preloads/pdfJsNativeOverrides.js"
+				...
+			}
+			...
+		}
+
+	IMPORTANT NOTE: If you set that path incorrectly it will cause Finsemble to stop working in that component.
+	Check your component's chrome console for the existence of FSBL. If it doesn't exist then check your path.
+ */
+
+/**
+ * This overrides the browser's built in window.open function by instead creating windows using LauncherClient.spawn.
+ * This ensures that the Finsemble workspace manager is aware of newly opened windows, that they can participate in
+ * the on screen workspace management, and that they can be restored with workspaces.
+ * 
+ * This version attempts to detect URLs pointing to PDF files and automatically opens them in the PDF viewer component.
+ */
+
+var originalWindowOpen = window.open;
+window.open = function (urlToOpen, name, specs, replace) {
+	// For some strange reason, openfin notifications use window.open. So we make an exception for that one case.
+	if (name && name.includes("openfin-child-window")) {
+		originalWindowOpen.call(window, urlToOpen, name, specs, replace);
+		return;
+	}
+
+	var parsedUrl = new URL(urlToOpen);
+	if (parsedUrl.pathname.toLowerCase().endsWith(".pdf")) {
+		//Handle a PDF
+		return handlePdfUrl(urlToOpen, name, specs, replace);
+	} else {
+		//Handle other URL
+		return handleOtherUrl(urlToOpen, name, specs, replace);
+	}
+}
+
+const handlePdfUrl = function(urlToOpen, name, specs, replace) {
```

Comment:
specs and replace params are unused / never read, are they needed?

Reply:
Left in as thats the signature of the built-in in JS function we're overloading